src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/subtle_alert_request.cc

```diff
@@ -0,0 +1,469 @@
+  if ((!(*message_)[strings::msg_params].keyExists(strings::alert_text1)) &&
+      (!(*message_)[strings::msg_params].keyExists(strings::alert_text2)) &&
+      (!(*message_)[strings::msg_params].keyExists(strings::tts_chunks) &&
+       (1 > (*message_)[strings::msg_params][strings::tts_chunks].length()))) {
+    LOG4CXX_ERROR(logger_, "Mandatory parameters are missing");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "Mandatory parameters are missing");
+    return false;
+  }
+
+  if ((*message_)[strings::msg_params].keyExists(strings::tts_chunks)) {
+    smart_objects::SmartObject& tts_chunks =
+        (*message_)[strings::msg_params][strings::tts_chunks];
+    mobile_apis::Result::eType verification_result =
+        MessageHelper::VerifyTtsFiles(tts_chunks, app, application_manager_);
+
+    if (mobile_apis::Result::FILE_NOT_FOUND == verification_result) {
+      LOG4CXX_ERROR(
+          logger_,
+          "MessageHelper::VerifyTtsFiles return " << verification_result);
+      SendResponse(false,
+                   mobile_apis::Result::FILE_NOT_FOUND,
+                   "One or more files needed for tts_chunks are not present");
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void SubtleAlertRequest::SendSubtleAlertRequest(int32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  ApplicationSharedPtr app = application_manager_.application(app_id);
+
+  smart_objects::SmartObject msg_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+  msg_params[hmi_request::alert_strings] =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
+
+  if ((*message_)[strings::msg_params].keyExists(strings::cancel_id)) {
+    msg_params[strings::cancel_id] =
+        (*message_)[strings::msg_params][strings::cancel_id].asInt();
+  }
+
+  int32_t index = 0;
+  if ((*message_)[strings::msg_params].keyExists(strings::alert_text1)) {
+    msg_params[hmi_request::alert_strings][index][hmi_request::field_name] =
+        hmi_apis::Common_TextFieldName::subtleAlertText1;
+    msg_params[hmi_request::alert_strings][index][hmi_request::field_text] =
+        (*message_)[strings::msg_params][strings::alert_text1];
+    index++;
+  }
+  if ((*message_)[strings::msg_params].keyExists(strings::alert_text2)) {
+    msg_params[hmi_request::alert_strings][index][hmi_request::field_name] =
+        hmi_apis::Common_TextFieldName::subtleAlertText2;
+    msg_params[hmi_request::alert_strings][index][hmi_request::field_text] =
+        (*message_)[strings::msg_params][strings::alert_text2];
+    index++;
+  }
+
+  // softButtons
+  if ((*message_)[strings::msg_params].keyExists(strings::soft_buttons)) {
+    msg_params[hmi_request::soft_buttons] =
+        (*message_)[strings::msg_params][strings::soft_buttons];
+    MessageHelper::SubscribeApplicationToSoftButton(
+        (*message_)[strings::msg_params], app, function_id());
+  } else {
```

Comment:
```suggestion
  }
  
  if ((*message_)[strings::msg_params].keyExists(strings::duration)) {
```

Adding duration parameter should not depend on whether or not softbuttons are included with the request.